src/useTreasury.ts

```diff
@@ -0,0 +1,57 @@
+import { Contract } from '@ethersproject/contracts';
+import { ImpactProviderContext } from './ImpactProvider';
+import { getContracts } from './contracts';
+import BaseERC20ABI from './abi/BaseERC20.json';
+import React from 'react';
+
+type TokenArgs = {
+    address: string;
+    name: string;
+};
+
+export const useTreasury = () => {
+    const { connection, address, provider } = React.useContext(ImpactProviderContext);
+
+    /**
+     * Get list of available community tokens
+     * @returns {Promise<{}[]>} array with name(symbol) and address of the available tokens
+     */
+    const getTokenList = async (): Promise<TokenArgs[]> => {
+        if (!connection || !address) {
+            throw new Error('No connection');
+        }
+
+        let token: TokenArgs;
+        const tokens: TokenArgs[] = [];
+
+        try {
+            const { treasury } = await getContracts(provider);
+            const listLength = await treasury.tokenListLength();
+
+            for (let index = 0; index < listLength; index++) {
+                token = await new Promise(resolve => {
```

Comment:
We can improve this to run in parallel with a Promise.all and make it load faster.